src/nvim/lua/converter.c

```diff
@@ -212,19 +212,27 @@ bool nlua_pop_typval(lua_State *lstate, typval_T *ret_tv)
           const char *s = lua_tolstring(lstate, -2, &len);
           if (cur.special) {
             list_T *const kv_pair = tv_list_alloc();
+
             tv_list_append_list(cur.tv->vval.v_list, kv_pair);
-            listitem_T *const key = tv_list_item_alloc();
-            *TV_LIST_ITEM_TV(key) = decode_string(s, len, kTrue, false, false);
-            tv_list_append(kv_pair, key);
-            if (TV_LIST_ITEM_TV(key)->v_type == VAR_UNKNOWN) {
+            typval_T s_tv = decode_string(s, len, kTrue, false, false);
+            if (s_tv.v_type == VAR_UNKNOWN) {
               ret = false;
               tv_list_unref(kv_pair);
```

Comment:
Is it okay to free the list `kv_pair` but keep it as an item in the list `cur.tv->vval.v_list`?